src/lib.rs

```diff
@@ -48,24 +36,15 @@ impl Plugin for JsScriptingPlugin {
             app.add_system_to_stage(
                 stage.clone(),
                 (move |world: &mut World| {
-                    let mut active = world.remove_non_send_resource::<ActiveScripts>().unwrap();
-                    world.resource_scope(|world, assets: Mut<Assets<JsScript>>| {
-                        let in_stage = active
-                            .by_stage
-                            .entry(stage.clone())
-                            .or_default()
-                            .iter()
-                            .filter(|&handle| assets.contains(handle));
-                        for handle in in_stage {
-                            if let Some(runtime) = active.runtimes.get_mut(handle) {
-                                if let RuntimeStatus::Loaded = runtime.status {
-                                    run_js_script(world, &mut runtime.runtime);
-                                }
-                            }
-                        }
-                    });
-
-                    world.insert_non_send_resource(active);
+                    let active_scripts = world.remove_resource::<ActiveScripts>().unwrap();
+                    let runtime = world.remove_non_send_resource::<JsRuntime>().unwrap();
+
+                    for script in &*active_scripts {
```

Comment:
> I think the exact init API will need to be well thought out but for now it is fine as is.

Good thing is it's easy to change. Also, just to demonstrate a pattern I used in my game, but didn't get to showing here yet, since the return value of `init()` is passed as the `this` object for every stage function call you can return a class, and store state in the class, which feels nice:

```ts
class Test1 implements ScriptSystems {
  counter: number = 0;

  update() {
    if (this.counter == 0) {
      info("First update for my script!");
    }

    trace(`Script update #${this.counter}`);
    this.counter++;
  }
}

/**
 * This is the only required function in scripts. It must return a type implementing
 * `ScriptSystems`.
 */
function init(): ScriptSystems {
  return new Test1();
}
```